# Some built in controllers, taken from
# https://github.com/ros-controls/roscon2022_workshop/blob/master/controlko_bringup/config/rrbot_controllers.yaml

controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    forward_position_controller:
      type: forward_command_controller/ForwardCommandController

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController


forward_position_controller:
  ros__parameters:
    joints:
      - front_left_hip_x
      - front_left_hip_y
      - front_left_knee
      - front_right_hip_x
      - front_right_hip_y
      - front_right_knee
      - rear_left_hip_x
      - rear_left_hip_y
      - rear_left_knee
      - rear_right_hip_x
      - rear_right_hip_y
      - rear_right_knee
      - arm_sh0
      - arm_sh1
      - arm_el0
      - arm_el1
      - arm_wr0
      - arm_wr1
      - arm_f1x
    interface_name: position

joint_trajectory_controller:
  ros__parameters:
    joints:
      - front_left_hip_x
      - front_left_hip_y
      - front_left_knee
      - front_right_hip_x
      - front_right_hip_y
      - front_right_knee
      - rear_left_hip_x
      - rear_left_hip_y
      - rear_left_knee
      - rear_right_hip_x
      - rear_right_hip_y
      - rear_right_knee
      - arm_sh0
      - arm_sh1
      - arm_el0
      - arm_el1
      - arm_wr0
      - arm_wr1
      - arm_f1x

    command_interfaces:
      - position

    state_interfaces:
      - position
      - velocity

    state_publish_rate: 50.0 # Defaults to 50
    action_monitor_rate: 20.0 # Defaults to 20

    allow_partial_joints_goal: false # Defaults to false
    constraints:
      stopped_velocity_tolerance: 0.01 # Defaults to 0.01
      goal_time: 0.0 # Defaults to 0.0 (start immediately)
